mono/metadata/w32file-unix.c

```diff
@@ -55,6 +55,13 @@
 #include "utils/strenc.h"
 #include "utils/refcount.h"
 
+#if 0 // FIXME icall-decl.h sometimes breaks things because of Boehm's handling of dlopen.
```

Comment:
Could you open a new GitHub issue about this.  Seems suspicious that handles headers would bring in GC-specific includes...

Reply:
Also it looks like `ves_icall_System_IO_DriveInfo_GetDriveType` should use `mono_dl_open` instead of `dlopen` directly.

Reply:
Agreed. That should provide for fixing the FIXME.
https://github.com/mono/mono/issues/11936